### ISSUE TITLE
[AVM Module feature]:  Storage module version definition modification

### ISSUE THREAD

**Issue by stredovek** [Azure/terraform-azurerm-avm-res-web-site#67]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Description

Is there plan to use different version definition  in this section:
``` terraform
module "avm_res_storage_storageaccount" {
  count = var.function_app_create_storage_account ? 1 : 0

  source  = "Azure/avm-res-storage-storageaccount/azurerm"
  **version = "0.1.1"**

  name                          = var.function_app_storage_account.name
  resource_group_name           = coalesce(var.function_app_storage_account.resource_group_name, var.resource_group_name)
  shared_access_key_enabled     = true
  public_network_access_enabled = true
  network_rules = {
    bypass         = ["AzureServices"]
    default_action = "Allow"
  }
  tags = var.tags
}
```
 something like  version ~> "0.1.1" ?
AVM storage module v0.1.1 has hardcoded /azure/azapi version 1.9.0 that is fixed in version 0.1.2. 
In my situation higher azapi version is needed in my other module and therefore terraform init fails.

**Comment by donovm4** [comment]
Hi @stredovek - per [TFFR1](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr1---category-composition---cross-referencing-modules) in the [AVM](https://azure.github.io/Azure-Verified-Modules/) spec, referencing to AVM modules MUST be pinned versions (i.e. "hardcoded"). 

I will run the e2e tests with the latest version of the `avm-res-storage-storageaccount` and once those tests pass, I will cut a release with the updates.

**Comment by donovm4** [comment]
addressed in v0.6.0 release

> NOTE: I see there is now a v0.1.3 for `avm-res-storage-storageaccount`. I will look into referencing latest version after testing.
